packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -4460,6 +4465,31 @@ export class ContainerRuntime
 		}
 	}
 
+	/**
+	 * Submit a Document Schema message to the Outbox.
+	 * In interactive clients, Document Schema messages submitted via this function will be
+	 * held in the Outbox until the next flush.
+	 * This avoids sending an op apart from user interaction, which is preferable when possible.
+	 *
+	 * @privateremarks
+	 * ID Allocation ops may prove to be well-suited to use this flow as well.
+	 * GC ops are not since they should be sent immediately.
+	 *
+	 * @param message - The Document Schema message to submit.
+	 */
+	private submitDocumentSchemaMessage(message: ContainerRuntimeDocumentSchemaMessage): void {
+		if (this.isSummarizerClient) {
+			this.submit(message);
+			return;
+		}
+
+		this.outbox.holdDocumentSchemaMessageUntilNextFlush({
+			runtimeOp: message,
+			referenceSequenceNumber: this.deltaManager.lastSequenceNumber,
```

Comment:
I think this will be a problem.  Batches must have one reference sequence number.

Two options:
1. Set `referenceSequenceNumber` from within flush to match the batch. This will be kind of a lie since the internals will have been generated at a prior ref seq (and in fact the message itself has a property `refSeq` which will be mismatched).  I think DocSchemaController can handle it, since it uses that `refSeq` for Compare-And-Swap, but it will be a bit surprising.
2. Move the whole check-if-we-need-to-upgrade-schema logic to run during Flush. I think I prefer this.

Reply:
are you planning to change this PR due to this?